roofit/roofitcore/src/RooAddPdf.cxx

```diff
@@ -265,40 +245,28 @@ RooAddPdf::RooAddPdf(const char *name, const char *title, const RooArgList& inPd
       _coefList.add(*rfrac) ;
 
       // In recursive mode we always have Ncoef=Npdf, since we added it just above
-      _haveLastCoef=kTRUE ;
+      _haveLastCoef=true ;
     }
 
   } else {
-    _haveLastCoef=kTRUE ;
+    _haveLastCoef=true ;
   }
 
-
-  _coefCache.resize(_pdfList.size());
-  _coefErrCount = _errorCount ;
-
-  TRACE_CREATE
+  finalizeConstruction();
 }
 
 
-
 ////////////////////////////////////////////////////////////////////////////////
 /// Generic constructor from list of extended PDFs. There are no coefficients as the expected
 /// number of events from each components determine the relative weight of the PDFs.
 ///
 /// All PDFs must inherit from RooAbsPdf.
 
 RooAddPdf::RooAddPdf(const char *name, const char *title, const RooArgList& inPdfList) :
-  RooAbsPdf(name,title),
-  _refCoefNorm("!refCoefNorm","Reference coefficient normalization set",this,kFALSE,kFALSE),
-  _refCoefRangeName(0),
-  _projectCoefs(kFALSE),
-  _projCacheMgr(this,10),
-  _pdfList("!pdfs","List of PDFs",this),
-  _coefList("!coefficients","List of coefficients",this),
-  _haveLastCoef(kFALSE),
-  _allExtendable(kTRUE),
-  _recursive(kFALSE)
+  RooAddPdf(name,title)
 {
+  _allExtendable = true;
```

Comment:
Consider moving this in member initialiser list?

Reply:
This would give a `warning: mem-initializer for ‘RooAddPdf::_allExtendable’ follows constructor delegation`.

Reply:
Ah, I missed that it's a delegating constructor! 👍